torchvision/models/detection/fcos.py

```diff
@@ -88,13 +88,12 @@ def compute_loss(
 
         # regression loss: GIoU loss
         # TODO: vectorize this instead of using a for loop
-        pred_boxes = [
-            self.box_coder.decode_single(bbox_regression_per_image, anchors_per_image)
-            for anchors_per_image, bbox_regression_per_image in zip(anchors, bbox_regression)
-        ]
+
+        pred_boxes = self.box_coder.decode_all(bbox_regression, anchors)
+
         # amp issue: pred_boxes need to convert float
         loss_bbox_reg = generalized_box_iou_loss(
-            torch.stack(pred_boxes)[foregroud_mask].float(),
+            pred_boxes[foregroud_mask].float(),
```

Comment:
nit: I may be wrong but it seems to me that `.float` is a bit redundant as `pred_boxes` will be float due to the way centers are computed.

Reply:
You are right!